### ISSUE TITLE
When using a backupstoragelocation not name default the logs get tons of spam about the default missing.

### ISSUE THREAD

**Issue by sharkymcdongles** [vmware-tanzu/velero#3228]
**What steps did you take and what happened:**
Create a velero deployment and name the only backupstoragelocation anything other than default.

**What did you expect to happen:**
No log spam.

**The output of the following commands will help us better understand what's going on**:
(Pasting long output into a [GitHub gist](https://gist.github.com) or other pastebin is fine.)

```
time="2021-01-11T17:58:50Z" level=warning msg="The specified default backup location named \"default\" was not found; for convenience, be sure to create one or make another backup location that is available the default" controller=backupstoragelocation logSource="pkg/controller/backupstoragelocation_controller.go:161"
time="2021-01-11T17:59:19Z" level=warning msg="The specified default backup location named \"default\" was not found; for convenience, be sure to create one or make another backup location that is available the default" controller=backupstoragelocation logSource="pkg/controller/backupstoragelocation_controller.go:161"
time="2021-01-11T17:59:19Z" level=warning msg="The specified default backup location named \"default\" was not found; for convenience, be sure to create one or make another backup location that is available the default" controller=backupstoragelocation logSource="pkg/controller/backupstoragelocation_controller.go:161"
time="2021-01-11T18:00:20Z" level=warning msg="The specified default backup location named \"default\" was not found; for convenience, be sure to create one or make another backup location that is available the default" controller=backupstoragelocation logSource="pkg/controller/backupstoragelocation_controller.go:161"
time="2021-01-11T18:00:20Z" level=warning msg="The specified default backup location named \"default\" was not found; for convenience, be sure to create one or make another backup location that is available the default" controller=backupstoragelocation logSource="pkg/controller/backupstoragelocation_controller.go:161"
time="2021-01-11T18:01:20Z" level=warning msg="The specified default backup location named \"default\" was not found; for convenience, be sure to create one or make another backup location that is available the default" controller=backupstoragelocation logSource="pkg/controller/backupstoragelocation_controller.go:161"
time="2021-01-11T18:01:20Z" level=warning msg="The specified default backup location named \"default\" was not found; for convenience, be sure to create one or make another backup location that is available the default" controller=backupstoragelocation logSource="pkg/controller/backupstoragelocation_controller.go:161"
time="2021-01-11T18:01:34Z" level=warning msg="The specified default backup location named \"default\" was not found; for convenience, be sure to create one or make another backup location that is available the default" controller=backupstoragelocation logSource="pkg/controller/backupstoragelocation_controller.go:161"
time="2021-01-11T18:02:20Z" level=warning msg="The specified default backup location named \"default\" was not found; for convenience, be sure to create one or make another backup location that is available the default" controller=backupstoragelocation logSource="pkg/controller/backupstoragelocation_controller.go:161"
time="2021-01-11T18:02:20Z" level=warning msg="The specified default backup location named \"default\" was not found; for convenience, be sure to create one or make another backup location that is available the default" controller=backupstoragelocation logSource="pkg/controller/backupstoragelocation_controller.go:161"
time="2021-01-11T18:03:20Z" level=warning msg="The specified default backup location named \"default\" was not found; for convenience, be sure to create one or make another backup location that is available the default" controller=backupstoragelocation logSource="pkg/controller/backupstoragelocation_controller.go:161"
time="2021-01-11T18:03:20Z" level=warning msg="The specified default backup location named \"default\" was not found; for convenience, be sure to create one or make another backup location that is available the default" controller=backupstoragelocation logSource="pkg/controller/backupstoragelocation_controller.go:161"
time="2021-01-11T18:04:21Z" level=warning msg="The specified default backup location named \"default\" was not found; for convenience, be sure to create one or make another backup location that is available the default" controller=backupstoragelocation logSource="pkg/controller/backupstoragelocation_controller.go:161"
time="2021-01-11T18:04:21Z" level=warning msg="The specified default backup location named \"default\" was not found; for convenience, be sure to create one or make another backup location that is available the default" controller=backupstoragelocation logSource="pkg/controller/backupstoragelocation_controller.go:161"
```

**Environment:**

- Velero version (use `velero version`): 1.5.2

**Comment by sharkymcdongles** [comment]
NM setting the config for the BSL to have:

profile: "default"

will solve I think from the docs. Will test and report back. Perhaps add this to the error message.

**Comment by sharkymcdongles** [comment]
Problem persists.

```
apiVersion: velero.io/v1
kind: BackupStorageLocation
metadata:
  annotations:
    meta.helm.sh/release-name: velero
    meta.helm.sh/release-namespace: velero
  creationTimestamp: "2021-01-11T17:21:30Z"
  generation: 2
  labels:
    app.kubernetes.io/instance: velero
    app.kubernetes.io/managed-by: Helm
    app.kubernetes.io/name: velero
    helm.sh/chart: velero-2.13.7
  name: gcp
  namespace: velero
  resourceVersion: "350459"
  selfLink: /apis/velero.io/v1/namespaces/velero/backupstoragelocations/gcp
  uid: b65761b8-ddca-460e-8368-8be46dc8d627
spec:
  config:
    profile: default
  objectStorage:
    bucket: backup
  provider: gcp
status:
  lastSyncedTime: "2021-01-11T20:13:51Z"
  lastValidationTime: "2021-01-11T20:13:52Z"
  phase: Available
```

**Comment by carlisia** [comment]
Hi @sharkymcdongles. In the soon to be released v1.6, we changed things so you will be able to set whichever BSL you want the default to be. The change was made here: https://github.com/vmware-tanzu/velero/pull/3092.

We also got rid of that message.

For now, there are two ways to make this warning go away:

1) Change the name of the BSL to "default". In the example above, your BSL is named :gcp", change that to "default".

2) Set the `--default-backup-storage-location` argument for the Velero server to be whatever you want the default BSL to be. In your case, it'd be "gcp". This will require a server restart.

I'm closing this issue, if you have any other question feel free to reopen it.
